cdk/src/open-data-platform/lambda/write-demographic-data-handler.ts

```diff
@@ -1,158 +1,223 @@
 // asset-input/src/open-data-platform/lambda/write-demographic-data-handler.js
+import {SecretsManager} from '@aws-sdk/client-secrets-manager';
+import createConnectionPool, {ConnectionPool, ConnectionPoolConfig, Queryable, sql} from '@databases/pg';
+import {bulkInsert, BulkOperationOptions} from '@databases/pg-bulk';
 import {APIGatewayEvent} from 'aws-lambda';
 
 const AWS = require('aws-sdk');
 const parse = require('csv-parser');
 const S3 = new AWS.S3();
-const rdsDataService = new AWS.RDSDataService();
+const secretsmanager = new SecretsManager({})
 
 const GEO_ID = 'GEOID';
 const RACE_TOTAL = 'RaceTotal';
 const WHITE_POPULATION = 'Estimate!!Total:!!White alone';
 const BLACK_POPULATION = 'Estimate!!Total:!!Black or African American alone';
 
-// Prepared SQL statements.
-const DELETE_DEMOGRAPHICS_TABLE =
-    'DELETE FROM demographics WHERE census_geo_id IS NOT NULL';
-
-// Reads the S3 CSV file and returns [DemographicsTableRow]s.
-const parseS3IntoDemographicsTableRow =
-    (s3Params: Object, numberRowsToWrite = 10): Promise<Array<DemographicsTableRow>> => {
-      const results: DemographicsTableRow[] = [];
-      return new Promise(function (resolve, reject) {
-        let count = 0;
-        const fileStream = S3.getObject(s3Params).createReadStream();
-        fileStream.pipe(parse())
-            .on('data',
-                (chunk) => {
-                  // Pause to allow for processing.
-                  fileStream.pause();
-                  // Only process 10 rows for now.
-                  if (count < numberRowsToWrite) {
-                    const row =
-                        new DemographicsTableRowBuilder()
-                            .censusGeoId(chunk[GEO_ID])
-                            .totalPopulation(parseInt(chunk[RACE_TOTAL]))
-                            .blackPopulation(parseInt(chunk[BLACK_POPULATION]))
-                            .whitePopulation(parseInt(chunk[WHITE_POPULATION]))
-                            .build();
-                    results.push(row);
-                  }
-                  count += 1;
-                  fileStream.resume();
-                })
-            .on('error',
-                (error) => {
-                  reject(error);
-                })
-            .on('end', () => {
-              console.log('Parsed ' + results.length + ' rows.');
-              resolve(results);
-            });
+// Establishes connection with DB with the given ARN.
+function connectToDb(secretArn: string): Promise<ConnectionPool> {
+  return new Promise(async function(resolve, reject) {
```

Comment:
I believe you can drop the use of a `new Promise` here by:
* Add the `async` keyword to this function declaration (`async function connectToDb`).
* Use `return pool` instead of `resolve(pool)`.
* Remove the try-catch blocks. `async will handle that for you, unless you want some specific error handling.

Reply:
Also, I saw you comment on slack about reusing the code from schema.handler.ts. Good idea.